pkg/codegen/schema/schema.go

```diff
@@ -1082,151 +1135,149 @@ type TypeSpec struct {
 	// A resource from this document is referenced as "#/resources/pulumi:type:token".
 	// A resource from another document is referenced as "path#/resources/pulumi:type:token", where path is of the form:
 	//   "/provider/vX.Y.Z/schema.json" or "pulumi.json" or "http[s]://example.com/provider/vX.Y.Z/schema.json"
-	Ref string `json:"$ref,omitempty"`
+	Ref string `json:"$ref,omitempty" yaml:"$ref,omitempty"`
 	// AdditionalProperties, if set, describes the element type of an "object" (i.e. a string -> value map).
-	AdditionalProperties *TypeSpec `json:"additionalProperties,omitempty"`
+	AdditionalProperties *TypeSpec `json:"additionalProperties,omitempty" yaml:"additionalProperties,omitempty"`
 	// Items, if set, describes the element type of an array.
-	Items *TypeSpec `json:"items,omitempty"`
+	Items *TypeSpec `json:"items,omitempty" yaml:"items,omitempty"`
 	// OneOf indicates that values of the type may be one of any of the listed types.
-	OneOf []TypeSpec `json:"oneOf,omitempty"`
+	OneOf []TypeSpec `json:"oneOf,omitempty" yaml:"oneOf,omitempty"`
 	// Discriminator informs the consumer of an alternative schema based on the value associated with it.
-	Discriminator *DiscriminatorSpec `json:"discriminator,omitempty"`
+	Discriminator *DiscriminatorSpec `json:"discriminator,omitempty" yaml:"discriminator,omitempty"`
 	// Plain indicates that when used as an input, this type does not accept eventual values.
-	Plain bool `json:"plain,omitempty"`
+	Plain bool `json:"plain,omitempty" yaml:"plain,omitempty"`
 }
 
 // DiscriminatorSpec informs the consumer of an alternative schema based on the value associated with it.
 type DiscriminatorSpec struct {
 	// PropertyName is the name of the property in the payload that will hold the discriminator value.
-	PropertyName string `json:"propertyName"`
+	PropertyName string `json:"propertyName" yaml:"propertyName"`
 	// Mapping is an optional object to hold mappings between payload values and schema names or references.
-	Mapping map[string]string `json:"mapping,omitempty"`
+	Mapping map[string]string `json:"mapping,omitempty" yaml:"mapping,omitempty"`
 }
 
 // DefaultSpec is the serializable form of extra information about the default value for a property.
 type DefaultSpec struct {
 	// Environment specifies a set of environment variables to probe for a default value.
-	Environment []string `json:"environment,omitempty"`
+	Environment []string `json:"environment,omitempty" yaml:"environment,omitempty"`
 	// Language specifies additional language-specific data about the default value.
-	Language map[string]json.RawMessage `json:"language,omitempty"`
+	Language map[string]RawMessage `json:"language,omitempty" yaml:"language,omitempty"`
 }
 
 // PropertySpec is the serializable form of an object or resource property.
 type PropertySpec struct {
-	TypeSpec
+	TypeSpec `yaml:",inline"`
```

Comment:
What does the `yaml:",inline"` do?

Reply:
It tells the YAML marshaler that the fields from `TypeSpec` will appear inline in the serialized form of `PropertySpec`. Without this, we'd need something like

```yaml
property:
    TypeSpec:
        type: string
```